src/video.cpp

```diff
@@ -243,9 +243,9 @@ int Video::open(String a_path, bool a_load_audio) {
 			av_hw_frame = av_frame_alloc();
 			sws_scale_frame(sws_ctx, av_hw_frame, av_frame);
 
-			y_data.resize(av_hw_frame->linesize[0] * resolution.y);
-			u_data.resize(av_hw_frame->linesize[1] * (resolution.y / 2));
-			v_data.resize(av_hw_frame->linesize[2] * (resolution.y / 2));
+			y_data = Image::create_empty(av_frame->linesize[0] , resolution.y, false, Image::FORMAT_R8);
+			u_data = Image::create_empty(av_frame->linesize[1] , resolution.y/2, false, Image::FORMAT_R8);
+			v_data = Image::create_empty(av_frame->linesize[2] , resolution.y/2, false, Image::FORMAT_R8);
 			padding = av_hw_frame->linesize[0] - resolution.x;
 
 			av_frame_unref(av_hw_frame);
```

Comment:
Don't we need to use av_hw_frame frame here as the data might be different after scaling the frame with sws_scale_frame? If that's not the case we could technically remove the sws_scale_frame and use av_frame for the padding I guess.